artifacts/definitions/Generic/Collectors/File.yaml

```diff
@@ -49,13 +56,16 @@ sources:
 
       -- Join all the collection rules into a single Glob plugin. This ensure we
       -- only make one pass over the filesystem. We only want LFNs.
-      LET hits = SELECT OSPath AS SourceFile, Size,
+      LET all_hits = SELECT OSPath AS SourceFile, Size,
                Btime AS Created,
                Ctime AS Changed,
                Mtime AS Modified,
                Atime AS LastAccessed
         FROM glob(globs=specs.Glob, accessor=Accessor)
         WHERE NOT IsDir AND log(message="Found " + SourceFile)
+    
+      LET size_hits = SELECT * FROM all_hits WHERE Size <= MaxFileSize     
+      LET hits = if(condition=MaxFileSize > 0, then=size_hits, else=all_hits)
```

Comment:
This will actually force the hits to materialize as it is a function - you should use something like
`SELECT * FROM if(condition = .... then=size_hits, else=all_hits)`

But in any case it is probably fine to just set MaxSize to be incredibly large by default and always include it rather than use if()

Also would be nice to report when a collection was skipped due to the size being too large:

```sql
WHERE NOT IsDir
    AND log(message="Found " + SourceFile)
    AND  ( Size <= MaxFileSize OR (log(message="Size exceeded ...") AND FALSE) )
```